csclient/csclient.go

```diff
@@ -257,28 +256,26 @@ func (s *Client) Publish(id *charm.URL, channels []params.Channel, resources map
 // It must be closed after use.
 type ResourceData struct {
 	io.ReadCloser
-	Revision int
```

Comment:
As long as the caller is not required to specify a revision, we must return the revision.